device/src/main/java/io/cloudthing/sdk/device/connectivity/mqtt/MqttCloudthingClientBuilder.java

```diff
@@ -33,7 +35,10 @@ public static MqttCloudthingClientBuilder getBuilder() {
 
     public IMqttCloudthingClient build() throws MqttException {
         validate();
-        MqttClient mqttClient = new MqttClient(getBrokerUri(), MqttClient.generateClientId(), new MemoryPersistence());
+        if (clientId == null) {
```

Comment:
I'd add also comparison with empty string (just to be sure)

Reply:
Yup, definietly